lib/src/messages/logical_replication_messages.dart

```diff
@@ -250,6 +252,10 @@ class RelationMessageColumn {
 }
 
 class RelationMessage implements LogicalReplicationMessage {
+  /// Stores the latest [RelationMessage]s for a given [relationId] to enable parsing of binary [TupleData]
+  /// from subsequent data modification messages.
+  static final Map<int, RelationMessage> _latest = {};
```

Comment:
If I understand this correctly, we don't really need the latest `RelationMessage`, only the `relationId -> typeOid` map is needed, let's not store more than needed.